onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -51,12 +51,18 @@ bool NodeGroupSelector::CheckQDQNodes(const GraphViewer& graph_viewer, const Nod
     return false;
   }
 
-  auto does_node_produce_graph_output = [&graph_viewer](const Node* node_ptr) {
-    return graph_viewer.NodeProducesGraphOutput(*node_ptr);
-  };
-
-  if (std::any_of(dq_nodes.begin(), dq_nodes.end(), does_node_produce_graph_output)) {
-    return false;
+  {
+    // a 'consumer' is an input of another node or a graph output
+    auto does_dq_node_have_multiple_consumers = [&graph_viewer](const Node* dq_node_ptr) {
+      // DQ has only one output to consider
+      const size_t output_edge_count = dq_node_ptr->GetOutputEdgesCount();
+      const bool produces_graph_output = graph_viewer.NodeProducesGraphOutput(*dq_node_ptr);
+      return produces_graph_output ? output_edge_count > 0 : output_edge_count > 1;
+    };
+
+    if (std::any_of(dq_nodes.begin(), dq_nodes.end(), does_dq_node_have_multiple_consumers)) {
```

Comment:
Ideally,  This shouldn't happen here, right?

Reply:
yeah, ideally it shouldn't. this is something of a sanity check.
at the moment, it's possible to skip running the DQ duplication transformer so I think this is possible.

Reply:
sorry, misunderstood the first comment. will move the check. may save the more extensive code restructuring for a later PR.

Reply:
moved check to NodeUnit constructor